compiler/rustc_abi/src/layout/ty.rs

```diff
@@ -175,6 +175,9 @@ pub trait TyAbiInterface<'a, C>: Sized + std::fmt::Debug {
     fn is_tuple(this: TyAndLayout<'a, Self>) -> bool;
     fn is_unit(this: TyAndLayout<'a, Self>) -> bool;
     fn is_transparent(this: TyAndLayout<'a, Self>) -> bool;
+    /// Returns `true` if the type is always passed indirectly. Currently only
+    /// used for `VaList`s.
+    fn is_pass_indirectly(this: TyAndLayout<'a, Self>) -> bool;
```

Comment:
I only see this as having an effect on x86-64 and aarch64, but doesn't the argument-passing algorithm already enforce this by making sure that VaList gets passed via Memory?

Reply:
And just to finish the thought: the difference is important (only) in an FFI context: a foreign `extern "C" {  fn(ap: VaList); }` would expect `ap` to be passed as `PassMode::Indirect { on_stack: false, .. }`

Reply:
huh.